i'm getting test timeouts on big uploads tonight.
```
[ERROR] Errors: 
[ERROR]   ITestAzureBlobFileSystemE2E.testOOBWritesAndReadFail:95 » TestTimedOut test ti...
[ERROR]   ITestAzureBlobFileSystemFlush.testWriteHeavyBytesToFileSyncFlush:160 » TestTimedOut
[ERROR]   ITestAbfsInputStream.testWithNoOptimization:58->createFileWithContent:179 » TestTimedOut
[ERROR]   ITestAbfsInputStreamReadFooter>ITestAbfsInputStream.testExceptionInOptimization:103->ITestAbfsInputStream.createFileWithContent:179 » TestTimedOut
[ERROR]   ITestAbfsInputStreamReadFooter.testSeekToBeginAndReadWithConfTrue:106->testSeekAndReadWithConf:163->seekReadAndTest:196 » TestTimedOut
[ERROR]   ITestAbfsInputStreamReadFooter.testSeekToEndAndReadWithConfTrue:146->testSeekAndReadWithConf:162->ITestAbfsInputStream.createFileWithContent:179 » TestTimedOut
[ERROR]   ITestAbfsInputStreamReadFooter>ITestAbfsInputStream.testWithNoOptimization:58->ITestAbfsInputStream.createFileWithContent:179 » TestTimedOut
[ERROR]   ITestAbfsInputStreamSmallFileReads>ITestAbfsInputStream.testExceptionInOptimization:103->ITestAbfsInputStream.createFileWithContent:179 » TestTimedOut
[ERROR]   ITestAbfsInputStreamSmallFileReads.testSeekToMiddleAndReadBigFileWithConfFalse:156->testSeekAndReadWithConf:166->ITestAbfsInputStream.createFileWithContent:179 » TestTimedOut
[INFO] 
```

earlier today i was uploading multiGB files to the same a/c, which was triggering throttling (laptop was plugged in to firewall). its now on wifi, so either its that or some throttling is still taking place

--- a/hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java
+++ b/hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java
@@ -806,15 +806,11 @@ public class AzureBlobFileSystemStore implements Closeable, ListingSupport {
         .map(c -> c.getBoolean(FS_AZURE_BUFFERED_PREAD_DISABLE, false))
         .orElse(false);
     // readahead is cluster config unless openfile explicitly sets it.
-    boolean readahead = abfsConfiguration.isReadAheadEnabled();
-    boolean readAheadEnabled = options
-            .map(c -> c.getBoolean(FS_AZURE_ENABLE_READAHEAD, readahead))
-            .orElse(readahead);
     return new AbfsInputStreamContext(abfsConfiguration.getSasTokenRenewPeriodForStreamsInSeconds())
             .withReadBufferSize(abfsConfiguration.getReadBufferSize())
             .withReadAheadQueueDepth(abfsConfiguration.getReadAheadQueueDepth())
             .withTolerateOobAppends(abfsConfiguration.getTolerateOobAppends())
-            .isReadAheadEnabled(readAheadEnabled)
+            .isReadAheadEnabled(abfsConfiguration.isReadAheadEnabled())
             .withReadSmallFilesCompletely(abfsConfiguration.readSmallFilesCompletely())
             .withOptimizeFooterRead(abfsConfiguration.optimizeFooterRead())
             .withReadAheadRange(abfsConfiguration.getReadAheadRange())

--- a/hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java
+++ b/hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java
@@ -147,7 +147,6 @@ import static org.apache.hadoop.fs.azurebfs.constants.AbfsHttpConstants.SINGLE_W
 import static org.apache.hadoop.fs.azurebfs.constants.AbfsHttpConstants.TOKEN_VERSION;
 import static org.apache.hadoop.fs.azurebfs.constants.ConfigurationKeys.AZURE_ABFS_ENDPOINT;
 import static org.apache.hadoop.fs.azurebfs.constants.ConfigurationKeys.FS_AZURE_BUFFERED_PREAD_DISABLE;
-import static org.apache.hadoop.fs.azurebfs.constants.ConfigurationKeys.FS_AZURE_ENABLE_READAHEAD;
 import static org.apache.hadoop.fs.azurebfs.constants.ConfigurationKeys.FS_AZURE_IDENTITY_TRANSFORM_CLASS;
 
 /**
@@ -805,7 +804,6 @@ public class AzureBlobFileSystemStore implements Closeable, ListingSupport {
     boolean bufferedPreadDisabled = options
         .map(c -> c.getBoolean(FS_AZURE_BUFFERED_PREAD_DISABLE, false))
         .orElse(false);
-    // readahead is cluster config unless openfile explicitly sets it.
     return new AbfsInputStreamContext(abfsConfiguration.getSasTokenRenewPeriodForStreamsInSeconds())
             .withReadBufferSize(abfsConfiguration.getReadBufferSize())
             .withReadAheadQueueDepth(abfsConfiguration.getReadAheadQueueDepth())

--- a/hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/constants/ConfigurationKeys.java
+++ b/hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/constants/ConfigurationKeys.java
@@ -188,7 +188,7 @@ public final class ConfigurationKeys {
   public static final String AZURE_KEY_ACCOUNT_SHELLKEYPROVIDER_SCRIPT = "fs.azure.shellkeyprovider.script";
 
   /**
-   * Enable or disable readahead buffer in AbfsInputStream. It is enabled by default.
+   * Enable or disable readahead buffer in AbfsInputStream.
    * Value: {@value}.
    */
   public static final String FS_AZURE_ENABLE_READAHEAD = "fs.azure.enable.readahead";

--- a/hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsInputStream.java
+++ b/hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsInputStream.java
@@ -748,7 +748,6 @@ public class AbfsInputStream extends FSInputStream implements CanUnbuffer,
   @VisibleForTesting
   public boolean isReadAheadEnabled() {
     return readAheadEnabled;
-
   }
 
   @VisibleForTesting

--- a/hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsInputStreamContext.java
+++ b/hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsInputStreamContext.java
@@ -80,7 +80,6 @@ public class AbfsInputStreamContext extends AbfsStreamContext {
     return this;
   }
 
-
   public AbfsInputStreamContext withReadAheadRange(
           final int readAheadRange) {
     this.readAheadRange = readAheadRange;

